artemis-server/src/main/java/org/apache/activemq/artemis/core/protocol/core/ServerSessionPacketHandler.java

```diff
@@ -159,7 +158,11 @@
 
    private final boolean direct;
 
-   private static final ThreadLocal<AtomicBoolean> inHandler = ThreadLocal.withInitial(AtomicBoolean::new);
+   //marker instance used to recognize if a thread is performing a packet handling
+   private static final Object DUMMY = Boolean.TRUE;
+
+   //a thread that has its thread-local map populated with DUMMY is performing a packet handling
+   private static final ThreadLocal<Object> inHandler = new ThreadLocal<>();
```

Comment:
Fair doosies. Maybe a nit if that's the case why not make it Object() this way stops any inference the boolean is being used as a true false flag. Which I'll be honest is what I was thinking this was.